packages/@lwc/engine-dom/src/renderer.ts

```diff
@@ -200,10 +190,8 @@ export function nextSibling(node: Node): Node | null {
 }
 
 export function attachShadow(element: Element, options: ShadowRootInit): ShadowRoot {
-    if (hydrating) {
-        return element.shadowRoot!;
-    }
-    return element.attachShadow(options);
+    const internals = (element as any).attachInternals?.();
```

Comment:
You can remove the `any` if you change the `element` type from `Element` to `HTMLElement`.

```ts
export function attachShadow(element: HTMLElement, options: ShadowRootInit): ShadowRoot {
    const internals = element.attachInternals?.();
```